client/src/App.js

```diff
@@ -10,20 +10,24 @@ import NavBar from './components/NavBar';
 import LandingPage from './pages/Landing/Landing';
 import FriendsLayout from './pages/Friends/Friends';
 import Dashboard from './pages/Dashboard/Dashboard';
+import Loading from './components/Loading';
 
 import './App.css';
 
 function App() {
   const user = React.useContext(UserContext);
   const friends = React.useContext(FriendsContext);
+  const [userLoading, setUserLoading] = useState(false);
 
   React.useEffect(() => {
     const fetchData = async () => {
       if (!user._id) {
-        await user.getCurrent();
-      }
-
-      if (user._id) {
+        setUserLoading(true);
+        await user.getCurrent((err) => {
+          setUserLoading(false);
+        });
+      } else {
+        setUserLoading(false);
```

Comment:
This works fine, but if we need the loading information in other components, you may want to add the userLoading state inside the UserContext, and then you can get the state from the context and do stuff with it

Reply:
I agree I think this may be helpful when fetching the profile page of individual users. But since this is not needed now, I will keep it as is.

Will refactor once we hit that case.